README.md

```diff
@@ -60,6 +60,29 @@ The profiling tool generates information which can be used for debugging and pro
 Information such as Spark version, executor information, properties and so on. This runs on either CPU or
 GPU generated event logs.
 
-Please refer to [spark qualification tool documentation](docs/spark-qualification-tool.md) 
+Please refer to [spark qualification tool documentation](docs/spark-qualification-tool.md)
 and [spark profiling tool documentation](docs/spark-profiling-tool.md)
-for more details on how to use the tools.
\ No newline at end of file
+for more details on how to use the tools.
+
+## Dependency for External Projects
+
+If you need to develop some functionality on top of RAPIDS Accelerator For Apache Spark (we currently
+limit support to GPU-accelerated UDFs) we recommend you declare our distribution artifact
+as a `provided` dependency.
```

Comment:
Curious why this is necessary?

Reply:
> Curious why this is necessary?
marking our artifact as provided is not  necessary, this is why I chose the word "recommend". It's mostly to prevent our jar and its dependency tree from being included into a customer's assembly jar if the customer chooses to use something like the shade plugin to build an uber jar

> Should we be marking our dist artifact's dependencies as provided instead
to me this is like not [having those dependencies at all](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope), and we utilize this maven feature  for the aggregator in this PR. If we declare say cudf  as provided in the dist pom, when the customer project directly depends on dist (even with the compile scope). cudf  would become its transitive dependency were it not declared provided. Maven drops provided transitive dependencies from the dependency tree.

Reply:
> It's mostly to prevent our jar and its dependency tree from being included into a customer's assembly jar if the customer chooses to use something like the shade plugin to build an uber jar

Thanks, that makes sense.  I just wanted to check there wasn't something else going on there.

Reply:
Given the Spark dependency is so complicated to communicate, does it make sense to leave that one off and only list the cudf dependency?  Seems silly to pull in Spark jars as part of pulling in the dist jar as a dependency if the Spark version has not bearing on what the user will use in practice.  I can see the pro of listing it, as then depending just on the dist jar will get you _some_ kind of test environment that should work for your supplied RAPIDS UDF.  The drawback is I could see users needing to go out of their way to exclude the spark dependencies from the dist jar so it doesn't conflict with the Spark version they actually care about when testing.

Reply:
> Given the Spark dependency is so complicated to communicate, does it make sense to leave that one off and only list the cudf dependency?  
I don't have a strong opinion on this

> Seems silly to pull in Spark jars as part of pulling in the dist jar as a dependency if the Spark version has not bearing on what the user will use in practice. 
just a reminder that the user can control this with -Dbuildver even though it's not user-friendly as we discussed with @tgravescs offline 

> I can see the pro of listing it, as then depending just on the dist jar will get you _some_ kind of test environment that should work for your supplied RAPIDS UDF.  The drawback is I could see users needing to go out of their way to exclude the spark dependencies from the dist jar so it doesn't conflict with the Spark version they actually care about when testing.

Looking at our udf-examples , and we don't require Spark to compile them necessarily (e.g. Scala UDF). Thus, we should indeed  drop it and leave it up to the customer declare it.